modules/templates/templates/templates.tsx

```diff
@@ -0,0 +1,250 @@
+    setIsModalOpen(false);
+    setTemplateToDelete(null);
+  };
+
+  const handleAddTemplate = async () => {
+    const newTemplate = {
+      name: "New Template",
+      preview: "New template preview",
+      createdAt: new Date().toISOString(),
+      fields: {},
+    };
+    const createdTemplate = await templateService?.createTemplate(newTemplate);
+
+    if (!createdTemplate) return;
+    setTemplates([...templates, createdTemplate]);
+    setEditingTemplate(createdTemplate);
+    messageHandler.handleSuccess(t("templates.createSuccess"));
+  };
+
+  const handleEdit = (template: GenesisTemplate) => {
+    setEditingTemplate(template);
+  };
+
+  const handleSave = async () => {
+    if (!editingTemplate) return;
+
+    const updatedTemplate = { ...editingTemplate };
+
+    // Ensure the date is in the correct format
+    if (updatedTemplate.createdAt) {
+      try {
+        const date = new Date(updatedTemplate.createdAt);
+        updatedTemplate.createdAt = date.toISOString();
+      } catch (error) {
+        messageHandler.handleError(t("templates.editError"));
+        return;
+      }
+      updatedTemplate.createdAt = new Date(
+        updatedTemplate.createdAt,
+      ).toISOString();
+    }
+
+    const savedTemplate = await templateService?.updateTemplate(
+      updatedTemplate.id,
+      updatedTemplate,
+    );
+    if (!savedTemplate) return;
+
+    setTemplates(
+      templates.map((t) => (t.id === savedTemplate.id ? savedTemplate : t)),
+    );
+    setEditingTemplate(null);
+    messageHandler.handleSuccess(t("templates.editSuccess"));
+  };
+
+  const handleInputChange = (field: keyof GenesisTemplate, value: string) => {
+    if (!editingTemplate) return;
+
+    setEditingTemplate({ ...editingTemplate, [field]: value });
+  };
+
+  const columns: ColumnConfig<GenesisTemplate>[] = [
+    {
+      title: t("templates.title"),
+      dataIndex: "name",
+      render: (template: GenesisTemplate) =>
+        editingTemplate && editingTemplate.id === template.id ? (
+          <input
+            type="text"
+            value={editingTemplate.name}
+            onChange={(e) => handleInputChange("name", e.target.value)}
+            className={styles.input}
+          />
+        ) : (
+          <div
+            onClick={() =>
+              router.push(`/app/template-detail?id=${template.id}`)
+            }
+            className="flex items-center"
+            style={{ gap: "1vh", cursor: "pointer" }}
+          >
+            <span>{template.name}</span>
+            <FaRegFolderOpen size={17.5} />
+          </div>
+        ),
+    },
+    {
+      title: t("templates.date"),
+      dataIndex: "createdAt",
+      render: (template: GenesisTemplate) =>
+        editingTemplate && editingTemplate.id === template.id ? (
+          <input
+            type="date"
+            value={
+              new Date(editingTemplate.createdAt).toISOString().split("T")[0]
```

Comment:
Can use `moment` library with `.format()` as the way you can format, more readible